connector/connect/common/src/main/scala/org/apache/spark/sql/connect/client/ArtifactManager.scala

```diff
@@ -179,6 +184,9 @@ class ArtifactManager(
     val responseHandler = new StreamObserver[proto.AddArtifactsResponse] {
       private val summaries = mutable.Buffer.empty[ArtifactSummary]
       override def onNext(v: AddArtifactsResponse): Unit = {
+        if (v.getSessionId != sessionId) {
+          throw new IllegalStateException(s"Session ID mismatch: $sessionId != ${v.getSessionId}")
+        }
```

Comment:
this would throw in the GRPC handler thread that calls this callback. I'm not sure what the effect of it would be...

Reply:
SparkConnectAddArtifactsHandler doesn't seem to be setting sessionid